src/CaptureClient/CaptureEventProcessorTest.cpp

```diff
@@ -549,6 +552,89 @@ TEST(CaptureEventProcessor, CanHandleGpuJobs) {
   EXPECT_EQ(hw_excecution_timer.user_data_key(), actual_hw_execution_key);
 }
 
+TEST(CaptureEventProcessor, CanHandleMemoryEventWrapper) {
+  MockCaptureListener listener;
+  auto event_processor =
+      CaptureEventProcessor::CreateForCaptureListener(&listener, std::filesystem::path{}, {});
+
+  ClientCaptureEvent event;
+  MemoryEventWrapper* memory_event_wrapper = event.mutable_memory_event_wrapper();
+  memory_event_wrapper->set_timestamp_ns(110);
```

Comment:
Could I ask for a comment explaining that we are using an average here and moving this line after setting the timestamp from all the MemoryUsage or where we are checking that?

Reply:
Done! Thanks for the suggestion!